### ISSUE TITLE
navigation-menu: Keyboard navigation does not work for list/simple variants [Accessibility]

### ISSUE THREAD

**Issue by ajitzero** [goetzrobin/spartan#1043]
### Please provide the environment you discovered this bug in.

Docs: https://www.spartan.ng/components/navigation-menu

### Which area/package is the issue in?

navigation-menu

### Description

For the "List", "Simple", and "With icon" variants, keyboard navigation does not work for focusing on menu items. Using Tab & Space keys works for the main "Component" variant, but not for the simple variants, where it focuses the entire menu items group, but doesn't go into it.

| Works | Doesn't work |
|--------|--------|
| <img width="807" height="653" alt="Working Menu" src="https://github.com/user-attachments/assets/37417d02-ece4-4936-a4b8-06b9e454a95f" /> | <img width="716" height="608" alt="Not working" src="https://github.com/user-attachments/assets/b0bec0ff-9cf2-4eba-b348-8cce6f3f8f51" /> |

### Please provide the exception or error you saw

```true

```

### I would be willing to submit a PR to fix this issue

- [ ] Yes
- [ ] No
